core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt

```diff
@@ -165,4 +170,273 @@ object GraphQLUtils {
         }
     }
 
+
+    data class GraphInfo(
+            var fields: MutableSet<String> = mutableSetOf(),
+            var edges: MutableSet<String> = mutableSetOf()//set to omit redundant nodes
+
+    )
+
+    data class LongestInfo(
+            var size: Int = 0,
+            var path: List<String> = listOf()
+    )
+
+    fun constructGraph(state: GraphQLActionBuilder.TempState, entryPoint: String, source: String, graph: MutableMap<String, GraphInfo>, history: MutableList<String>, objectFieldsHistory: MutableSet<String>): MutableMap<String, GraphInfo> {
+
+        for (element in state.tables) {
+            if (element.tableType?.toLowerCase() != entryPoint.toLowerCase()) {
+                continue
+            }
+
+            if (entryPoint.toLowerCase() == "query" || entryPoint.toLowerCase() == "querytype" || entryPoint.toLowerCase() == "root" || entryPoint.toLowerCase() == "mutation") {//First entry point
+
+                if (element.kindOfTableFieldType.toString().toLowerCase() == "object") {
```

Comment:
`element.kindOfTableFieldType.toString().toLowerCase()` is called several times. can put in a variable.
(I remember doing something like this once with you... was it in this very file but different branch? )

Reply:
-Done